src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Parser {
+    private Ui ui;
+
+    /**
+     * Parses inputs provided by ui, and directs ui to print corresponding output.
+     * @param ui ui used to interact with users.
+     */
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Executes action associated with provided string input, and prints out relevant details to console.
+     * @param input command associated with an action
+     * @param taskList storage used to carry out actions
+     * @return whether Duke has completed executing
+     */
+    public boolean parse(String input, TaskList taskList) {
+        Pattern todoPattern = Pattern.compile("todo (.*)");
+        Pattern deadlinePattern = Pattern.compile("deadline (.*) /by (.*)");
+        Pattern eventPattern = Pattern.compile("event (.*) /at (.*)");
+        Pattern findPattern = Pattern.compile("find (.*)");
+
+        // print out list
```

Comment:
Perhaps it would be better if the first word of the comment was capitalised?

Reply:
Amended as of 4030096e42b9fe4bd5c9ccecfed49e72b055c620